builder/metadata.go

```diff
@@ -264,7 +279,7 @@ func (b *Builder) UpdateVersions(nextMix, nextUpstream uint32) error {
 	var latest uint32
 	var err error
 
-	checkUpstream := (nextUpstream != b.UpstreamVerUint32)
+	checkUpstream := (nextUpstream != b.UpstreamVerUint32) && !Offline
```

Comment:
While this logic makes sense, it's becoming clear we have a growing issue with readability & code structure by needing these kinds of conditional statements. It would be useful to go over the code and see how we could restructure things to not need complex conditionals for things like offline. An option would be to move the offline check further into the actual function being called, i.e getUpstreamFormatRange().

Reply:
I agree. Offline has the same issue that Native had before my CLI patch. In my opinion, the ideal solution would be isolating all commands that require network access under a separate network module and handle lack of network in there. This would make online/offline transparent for the rest of the code, pretty much like Native is right now.
With that said, I'd rather change this after we get offline working since it will is a pretty significant change to the code.